fetch/fetch.go

```diff
@@ -499,7 +501,7 @@ func (f *Fetch) failAfterRetry(hash types.Hash32) {
 	}
 
 	// first check if we have it locally from gossips
-	if _, err := f.bs.Get(req.hint, hash.Bytes()); err == nil {
+	if has, err := f.bs.Has(req.hint, hash.Bytes()); err == nil && has {
```

Comment:
i agree that it is much better, do you mind submitting this change separately?

Reply:
I opened a separate PR: https://github.com/spacemeshos/go-spacemesh/pull/5602